src/ec2/sshKeyPair.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as fs from 'fs-extra'
+import { ToolkitError } from '../shared/errors'
+import { ChildProcess } from '../shared/utilities/childProcess'
+
+export class SshKeyPair {
+    private publicKeyPath: string
+    private constructor(keyPath: string) {
+        this.publicKeyPath = `${keyPath}.pub`
+    }
+
+    public static async getSshKeyPair(keyPath: string) {
+        const keyExists = await fs.pathExists(keyPath)
+        if (!keyExists) {
+            await SshKeyPair.generateSshKeyPair(keyPath)
+        }
+        return new SshKeyPair(keyPath)
+    }
+
+    public static async generateSshKeyPair(keyPath: string): Promise<void> {
+        const process = new ChildProcess(`ssh-keygen`, ['-t', 'rsa', '-N', '', '-q', '-f', keyPath])
+        const result = await process.run()
+        if (result.exitCode !== 0) {
+            throw new ToolkitError('ec2: Failed to generate ssh key')
+        }
+    }
+
+    public getPublicKeyPath(): string {
+        return this.publicKeyPath
+    }
+
+    public async getPublicKey(): Promise<string> {
+        const contents = await fs.readFile(this.publicKeyPath, 'utf-8')
+        return contents
```

Comment:
Are callers expected to handle any exceptions if the public key is deleted while the extension is running? Also, use `SystemUtilities` for fs calls, it'll automatically display permissions issues.

Reply:
The caller doesn't necessarily need to handle the exception. We just want to make sure that if this ever does happen, it should be obvious what went wrong. That might be as simple as re-throwing the error with extra information.